### PR TITLE
UPSTREAM: <carry>: openshift: Use location from API over secret and unit test

### PR DESCRIPTION
**Pull request by enxebre**
Use location from API over secret and unit test. In follow ups we might want to drop location and resource group from the secret

### CODE REVIEW

**Comment by ingvagabund** [comment]
Why do we need to use the location from the machine config? What's the location field for?

**Comment by enxebre** [comment]
> Why do we need to use the location from the machine config? What's the location field for?
That's simply how a consumer choose the machine region

**Comment by ingvagabund** [comment]
Is `Location` Azure keyword for a region? If not, can't we just rename `Location` to `Region`?

**Comment by enxebre** [comment]
> Is Location Azure keyword for a region? If not, can we just rename Location to Region?

Yeh, this is consistent with upstream and azure API https://github.com/Azure/azure-sdk-for-go/blob/master/services/compute/mgmt/2019-03-01/compute/models.go#L692-L697 and https://github.com/Azure/azure-sdk-for-go/blob/master/services/compute/mgmt/2019-03-01/compute/models.go#L6508

File: pkg/cloud/azure/actuators/machine_scope_test.go [hunk 1]
```diff
@@ -169,3 +170,69 @@ func TestCredentialsSecretFailures(t *testing.T) {
...
+					"azure_region":          []byte("dummyRegion"),
+					"azure_resource_prefix": []byte("dummyClusterName"),
+				},
+			},
+			expectedLocation:      "test",
+			expectedResourceGroup: "test",
+		},
+	}
+
+	for _, tc := range testCases {
```
Comment (ingvagabund) [inline pkg/cloud/azure/actuators/machine_scope_test.go]: There is only one test case. No need for table driven testing. What does `TestNewMachineScope` mean? Are you testing the location set in machine's provider config overrides the one in the secret? If so, choose different locations in each object so you can validate which has higher priority. Also important to test other cases when the location in the secret is not set. Or, location in the machine provider config is not set.

**Pull request merged**